nap/fields.py

```diff
@@ -113,10 +113,30 @@ def descrub_value(self, val, for_read=False):
 
 class DecimalField(Field):
     def scrub_value(self, val):
-        return Decimal(str(val)) if val else None
+        """Return a Decimal object using the provided value.
+
+        If the provided value is None or empty return None instead
+        of returning a Decimal.
+        """
+        if val is None:
+            return None
+
+        val = str(val)
+
+        return Decimal(val) if val else None
```

Comment:
I was initially confused by the same thing. Maybe that could be made more clear?